relay-metrics/src/aggregatorservice.rs

```diff
@@ -128,157 +94,268 @@ pub struct FlushBuckets {
     pub buckets: Vec<Bucket>,
 }
 
+/// Parameters used by the [`AggregatorService`].
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(default)]
+pub struct AggregatorServiceConfig {
+    /// Parameters used by the [`Aggregator`].
+    #[serde(flatten)]
+    pub aggregator: AggregatorConfig,
+    /// The approximate maximum number of bytes submitted within one flush cycle.
+    ///
+    /// This controls how big flushed batches of buckets get, depending on the number of buckets,
+    /// the cumulative length of their keys, and the number of raw values. Since final serialization
+    /// adds some additional overhead, this number is approxmate and some safety margin should be
+    /// left to hard limits.
+    pub max_flush_bytes: usize,
+
+    /// Maximum amount of bytes used for metrics aggregation.
+    ///
+    /// When aggregating metrics, Relay keeps track of how many bytes a metric takes in memory.
+    /// This is only an approximation and does not take into account things such as pre-allocation
+    /// in hashmaps.
+    ///
+    /// Defaults to `None`, i.e. no limit.
+    pub max_total_bucket_bytes: Option<usize>,
+
+    /// The number of logical partitions that can receive flushed buckets.
+    ///
+    /// If set, buckets are partitioned by (bucket key % flush_partitions), and routed
+    /// by setting the header `X-Sentry-Relay-Shard`.
+    pub flush_partitions: Option<u64>,
+}
+
+impl Default for AggregatorServiceConfig {
+    fn default() -> Self {
+        Self {
+            max_flush_bytes: 5_000_000, // 5 MB
+            flush_partitions: None,
+            max_total_bucket_bytes: None,
+            aggregator: AggregatorConfig::default(),
+        }
+    }
+}
+
+/// Contains an [`AggregatorServiceConfig`] for a specific scope.
+///
+/// For now, the only way to scope an aggregator is by [`MetricNamespace`].
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ScopedAggregatorConfig {
+    /// Name of the aggregator, used to tag statsd metrics.
+    pub name: String,
+    /// Condition that needs to be met for a metric or bucket to be routed to a
+    /// secondary aggregator.
+    pub condition: Condition,
+    /// The configuration of the secondary aggregator.
+    pub config: AggregatorServiceConfig,
```

Comment:
Shouldn't this be `AggregatorConfig` (and this type be moved to the `aggregator` module)?
```suggestion
    /// The configuration of the secondary aggregator.
    pub config: AggregatorConfig,
```

Reply:
I thought there would be some compatiblity questions, but since not then Ill just change it to aggregatorconfig